pkg/csi/service/wcpguest/controller.go

```diff
@@ -280,9 +286,27 @@ func (c *controller) CreateVolume(ctx context.Context, req *csi.CreateVolumeRequ
 		if err != nil {
 			if errors.IsNotFound(err) {
 				diskSize := strconv.FormatInt(volSizeMB, 10) + "Mi"
-				claim := getPersistentVolumeClaimSpecWithStorageClass(supervisorPVCName,
-					c.supervisorNamespace, diskSize, supervisorStorageClass, getAccessMode(accessMode))
+				var annotations map[string]string
+				// Generate Annotation from req.GetAccessibilityRequirements().Preferred, refer to divyens csi-provisioner change
+				// Update getPersistentVolumeClaimSpecWithStorageClass to take in annotation as a param.
+				if commonco.ContainerOrchestratorUtility.IsFSSEnabled(ctx, common.TKGsHA) {
+					// generate new annotations.
```

Comment:
Only execute this code block when `req.AccessibilityRequirements` is not nil after checking the feature gate.